torch/_inductor/kernel/bmm.py

```diff
@@ -34,39 +34,52 @@ def bmm_grid(b, m, n, meta):
     stride_bk = {{stride("B", 1)}}
     stride_bn = {{stride("B", 2)}}
 
-    # based on triton.ops.matmul
-    pid = tl.program_id(0)
-    grid_m = (M + BLOCK_M - 1) // BLOCK_M
-    grid_n = (N + BLOCK_N - 1) // BLOCK_N
-
     # re-order program ID for better L2 performance
-    width = GROUP_M * grid_n
-    group_id = pid // width
-    group_size = min(grid_m - group_id * GROUP_M, GROUP_M)
-    pid_m = group_id * GROUP_M + (pid % group_size)
-    pid_n = (pid % width) // (group_size)
-
-    rm = pid_m * BLOCK_M + tl.arange(0, BLOCK_M)
-    rn = pid_n * BLOCK_N + tl.arange(0, BLOCK_N)
-    ram = tl.max_contiguous(tl.multiple_of(rm % M, BLOCK_M), BLOCK_M)
-    rbn = tl.max_contiguous(tl.multiple_of(rn % N, BLOCK_N), BLOCK_N)
-    rk = tl.arange(0, BLOCK_K)
+    pid = tl.program_id(0)
+    num_pid_m = tl.cdiv(M, BLOCK_M)
+    num_pid_n = tl.cdiv(N, BLOCK_N)
+
+    num_pid_in_group = GROUP_M * num_pid_n
+    group_id = pid // num_pid_in_group
+    first_pid_m = group_id * GROUP_M
+    GROUP_M = min(num_pid_m - first_pid_m, GROUP_M)
+    pid_m = first_pid_m + (pid % GROUP_M)
+    pid_n = (pid % num_pid_in_group) // GROUP_M
+    block_offset_m = pid_m * BLOCK_M
+    block_offset_n = pid_n * BLOCK_N
 
     idx_q = tl.program_id(1)  # batch dimension for BMM
-    A = A + (ram[:, None] * stride_am + rk[None, :] * stride_ak + idx_q*stride_aq)
-    B = B + (rk[:, None] * stride_bk + rbn[None, :] * stride_bn + idx_q*stride_bq)
+
+    A_block_ptr = tl.make_block_ptr(
+        base=A + idx_q * stride_aq,
+        shape=(M, K),
+        strides=(stride_am, stride_ak),
+        offsets=(block_offset_m, 0),
+        block_shape=(BLOCK_M, BLOCK_K),
+        order=(1, 0),
```

Comment:
`order=` should not be hardcoded

Reply:
Hmm, this depends on the layout. I guess I can check if stride_am > stride_ak?

Reply:
You could add a stride_order helper similar to:
https://github.com/pytorch/pytorch/blob/13af952f94e611f67ac306233898b753a9cf73d2/torch/_inductor/select_algorithm.py#L205

Something like:
```
sym_strides = self.named_input_nodes[name].get_stride()
assert len(sym_strides) == 2
if V.graph.sizevars.evaluate_expr(sympy.Lt(sym_strides[0], sym_strides[1])):
  return "(0, 1)"
else:
  return "(1, 0)"
```